Automattic/wp-desktop
# feat(logging): replace debug with custom logger

# [2] Replace Debug With Custom Logger

## Description

This PR is a follow-on to #814 and replaces the `debug` module with the custom logger implementation. Existing logging items are converted from the usage of `debug` to `lib/logger`.

## Highlights

Conversion of existing log items is fairly straightforward. For the preload script, however, log items are transmitted via IPC from the renderer to the main process. This is to avoid use of the `remote` module, which is regarded as highly insecure (and in fact will be excluded from future versions of Electron starting with v9).

## To Test

Build and run the application (or download [built artifacts for this PR](https://app.circleci.com/pipelines/github/Automattic/wp-desktop/9418/workflows/a024e7b4-6b65-4d67-a304-f64b5c8d1b11/jobs/56585)). Verify application logs are persisted to the following locations on each platform:

_Note_: Mac artifact is not notarized (notarized on tag only), so will require opening via System Preferences -> Security.

  - Mac: `$HOME/Library/Application Support/WordPress.com/logs`
  - Linux: `/home/<username>/.config/WordPress.com/logs`
  - Windows 10: `C:\Users\<username>\AppData\Roaming\WordPress.com\logs`
